### DISCUSSION TITLE
customHandlebarsExtensionFile ignored when running GenerateReleaseNotesConsoleTester.js

### DISCUSSION THREAD

**Discussion by mlop3s** [rfennell/AzurePipelines#954] [Q&A] [answered]
Hi,

i've written some extensions for handlebars and they work on the azure pipeline, however when i try to load them as a file it does not work.
I've tried using the test command line by adding 
```
    "customHandlebarsExtensionFile": "customHandlebarsExtension",
    "customHandlebarsExtensionFolder": "C:/Temp/",
```
The file is in C:\Temp and it's named "customHandlebarsExtension.js" As documented, i skip the extension.
- I've actually tried with and without.

The functions I've added are not recognized.
##vso[task.issue type=error;]Error Processing handlebars [Error: Missing helper: "hasDbScripts"]

However, if I add them directly, like:
```
    "customHandlebarsExtensionCode": "module.exports = {\r\n    hasDbScripts: function (commits) {",
```

It works.
What am I missing?

Cheers,
Marco

**Comment by rfennell** [comment, marked as answer]
Currently, the `customHandlebarsExtensionCode` is the only way to pass in a custom extension.

`customHandlebarsExtensionFile` is just the file name used to store the extension code at runtime.

I see that this is confusing. I added an issue #955  to change the code so that you can pass a file name.

**Reply by mlop3s** [reply]
Great. Thanks.
Awesome extension.

**Reply by rfennell** [reply]
I have made the change it is in 3.44.4 and it is in the process of being released now
